posthog/temporal/workflows/s3_batch_export.py

```diff
@@ -0,0 +1,342 @@
+import datetime as dt
+import json
+from dataclasses import dataclass
+from string import Template
+
+from aiohttp import ClientSession
+from django.conf import settings
+from temporalio import activity, workflow
+from temporalio.common import RetryPolicy
+
+from posthog.settings.base_variables import DEBUG, TEST
+from posthog.temporal.workflows.base import (
+    CreateBatchExportRunInputs,
+    PostHogWorkflow,
+    UpdateBatchExportRunStatusInputs,
+    create_export_run,
+    update_export_run_status,
+)
+
+INSERT_INTO_S3_QUERY_TEMPLATE = Template(
+    """
+    INSERT INTO FUNCTION s3({path}, $auth {file_format})
```

Comment:
Also, to register my objections, I would rather we handle the S3 upload ourselves than rely on ClickHouse here. The removal of a hop is nice and all, but debugging things in ClickHouse has been tricky, and I'd take slow and steady here over a marginal reduction in network traffic.

It also makes the management of ClickHouse a little annoying as we're now coupling the app functionality with ClickHouse config for not enough gain imo.

Reply:
(But we can change this outside of this PR)

Reply:
A fair objection 👍

If this becomes becomes an issue in the future more than happy to do the extra work. Very little would need to be deleted as we still need to select the data, configure the destination, schedule the workflow... So I think we can build on top of this as we discover things